component/src/main/java/org/wso2/carbon/identity/authenticator/LinkedIn/LinkedInAuthenticator.java

```diff
@@ -0,0 +1,350 @@
+                                                 HttpServletResponse response, AuthenticationContext context)
+            throws AuthenticationFailedException {
+        try {
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            String clientId = authenticatorProperties.get(LinkedInAuthenticatorConstants.CLIENT_ID);
+            String clientSecret = authenticatorProperties.get(LinkedInAuthenticatorConstants.CLIENT_SECRET);
+            String tokenEndPoint = getTokenEndpoint(authenticatorProperties);
+            if (tokenEndPoint == null) {
+                tokenEndPoint = authenticatorProperties.get(LinkedInAuthenticatorConstants.OAUTH2_TOKEN_URL);
+            }
+            String callbackurl = getCallbackUrl(authenticatorProperties);
+            OAuthAuthzResponse authzResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+            String code = authzResponse.getCode();
+            OAuthClientRequest accessRequest;
+            try {
+                accessRequest = OAuthClientRequest.tokenLocation(tokenEndPoint)
+                        .setGrantType(GrantType.AUTHORIZATION_CODE)
+                        .setClientId(clientId).setClientSecret(clientSecret)
+                        .setRedirectURI(callbackurl).setCode(code)
+                        .buildBodyMessage();
+            } catch (OAuthSystemException e) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Exception while building request for request access token", e);
+                }
+                throw new AuthenticationFailedException(e.getMessage(), e);
+            }
+            // create OAuth client that uses custom http client under the hood
+            OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+            OAuthClientResponse oAuthResponse;
+            try {
+                oAuthResponse = oAuthClient.accessToken(accessRequest);
+            } catch (Exception e) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Exception while requesting access token", e);
+                }
+                throw new AuthenticationFailedException(e.getMessage(), e);
+            }
+            AuthenticatedUser authenticatedUserObj;
+            String accessToken = oAuthResponse.getParam(LinkedInAuthenticatorConstants.ACCESS_TOKEN);
+            if (accessToken != null) {
+                Map<ClaimMapping, String> claims;
+                Map<String, Object> userClaims = getUserClaims(oAuthResponse);
+                if (userClaims != null && !userClaims.isEmpty()) {
+                    authenticatedUserObj = AuthenticatedUser
+                            .createFederateAuthenticatedUserFromSubjectIdentifier(String.valueOf(userClaims.get("id")));
+                    authenticatedUserObj
+                            .setAuthenticatedSubjectIdentifier(String.valueOf(userClaims.get("lastName")));
```

Comment:
Moved to constant file https://github.com/wso2-extensions/identity-outbound-auth-linkedIn/pull/3